File: Stacks - Java/StackAsArray.java
public class StackAsArray {

    /**
     * Init a constant which represents the capacity of the stack
     */
    private static final int n = 5;

    /**
     * Init an array 'stack' and take previously declared 'n' as size
     */
    private int[] stack = new int[n];

    /**
     * Init the variable and set it to '-1'
     */
    private int top = -1;

    /**
     * init the variable to show current items in stack
     */
    private int currentItems = 0;

    /**
     * Inserts a value in the stack
     * @param pushItem the value to be inserted
     */
    public void push(int pushItem) {
        if(isFull()) System.out.println("The stack is full");
        else {
            top++;
            stack[top] = pushItem;
            currentItems++;
        }
    }

    /**
     * Deletes a value in the stack
     */
    public void pop() {
        if(isEmpty()) System.out.println("The stack is empty");
        else {
            top--;
            currentItems--;
        }
    }

    /**
     * Shows the top item of the stack
     */
    public void peek() {
        if(isEmpty()) System.out.println("The stack is empty");
        else {
            System.out.println(stack[top] + " -> top");
        }
    }

    /**
     * Shows the stack
     */
    public void display() {
        for(int i = top; i >= 0; i--) {
            System.out.println(stack[i]);
        }
    }

    /**
     * Checks if the stack is full
     * @return the full stack-condition
     */
    public boolean isFull() {
      return top == n-1;
    }

    /**
     * Checks if the stack is empty
     * @return the empty stack-condition
     */
    public boolean isEmpty() {
        return top == -1;
    }

    /**
     * Shows the amount of items in stack
     */
    public void getSize() {
        System.out.println(currentItems + " -> current Values");
    }
}

File: Stacks - Python/StackAsArray.py
class StackAsArray:

    # Init the variables and the array
    def __init__(self):
        self.n = 5
        self.stack = [0] * self.n
        self.top = -1
        self.current_items = 0

    # Inserts a value in the stack
    def push(self, push_item):
        if self.is_full():
            print("The stack is full")
        else:
            self.top += 1
            self.stack[self.top] = push_item
            self.current_items += 1

    # Deletes a value in the stack
    def pop(self):
        if self.is_empty():
            print("The stack is empty")
        else:
            self.top = self.top - 1
            self.current_items -= 1

    # Shows the top item of the stack
    def peek(self):
        if self.is_empty():
            print("The stack is empty")
        else:
            print(f"{self.stack[self.top]} -> top")

    # Shows the stack
    def display(self):
        for i in range(self.top, -1, -1):
            print(self.stack[i])

    # Checks if the stack is full
    def is_full(self):
        return self.top == self.n - 1

    # Checks if the stack is empty
    def is_empty(self):
        return self.top == -1

    # Shows the amount of items in stack
    def get_size(self):
        print(f"{self.current_items} -> current Values")


# The main method
def main():
    stack = StackAsArray()

    stack.push(4)
    stack.push(3)
    stack.push(12)
    # Output at this point (as array): [12, 3, 4]

    stack.pop()
    # Output by now (as array): [3, 4]

    stack.push(8)
    stack.push(10)
    stack.push(9)
    # Output by now (as array): [9, 10, 8, 3, 4]

    stack.peek()
    stack.get_size()
    stack.display()
    # The Output will be (as array): [9, 10, 8, 3, 4], 9 -> top, 5 -> current Values


# Used for the main method
if __name__ == "__main__":
    main()
